Examples/GenerativeAISample/FunctionCallingSample/Screens/FunctionCallingScreen.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import GenerativeAIUIComponents
+import GoogleGenerativeAI
+import SwiftUI
+
+struct FunctionCallingScreen: View {
+  @EnvironmentObject
+  var viewModel: FunctionCallingViewModel
+
+  @State
+  private var userPrompt = "What is 100 Euros in U.S. Dollars?"
+
+  enum FocusedField: Hashable {
+    case message
+  }
+
+  @FocusState
+  var focusedField: FocusedField?
+
+  var body: some View {
+    VStack {
+      ScrollViewReader { scrollViewProxy in
+        List {
+          Text("Interact with a currency conversion API using function calling in Gemini.")
+          ForEach(viewModel.messages) { message in
+            MessageView(message: message)
+          }
+          if let error = viewModel.error {
+            ErrorView(error: error)
+              .tag("errorView")
+          }
+        }
+        .listStyle(.plain)
+        .onChange(of: viewModel.messages, perform: { newValue in
+          if viewModel.hasError {
+            // wait for a short moment to make sure we can actually scroll to the bottom
```

Comment:
nit: capitalization and period - here and below.

Reply:
Done